packages/core/src/amazonqFeatureDev/controllers/chat/messenger/messenger.ts

```diff
@@ -46,7 +46,20 @@ export class Messenger {
         )
     }
 
-    public sendErrorMessage(errorMessage: string, tabID: string, retries: number, phase?: SessionStatePhase) {
+    public sendErrorMessage(
+        errorMessage: string,
+        tabID: string,
+        retries: number,
+        phase?: SessionStatePhase,
+        monthlyLimitError: boolean = false
+    ) {
+        if (monthlyLimitError) {
```

Comment:
Instead of adding another code path to this function I would probably create a separate function that just handles taking in a string (+ other arguments as required) and then sending the error message itself. That way we don't have to complicate this function signature more and it reduces potential code paths

Reply:
As discussed offline, creating separate method explicitly for rendering monthlyQuota limit error; we can later update method name if required to handle for other use-cases.